lib/script.ex

```diff
@@ -0,0 +1,669 @@
+  end
+
+  @doc """
+  	raw_combine directly concatenates two scripts with no checks.
+  """
+  @spec raw_combine(t(), t()) :: t()
+  def raw_combine(%__MODULE__{items: s1}, %__MODULE__{items: s2}),
+    do: %__MODULE__{items: s1 ++ s2}
+
+  @doc """
+  	display_script returns a human readable string of the script, with
+  	op_codes shown by name rather than number. 
+  """
+  @spec display_script(t()) :: String.t()
+  def display_script(script) do
+    " " <> scriptxt = display_script(script, "")
+    scriptxt
+  end
+
+  defp display_script(%__MODULE__{items: []}, acc), do: acc
+
+  defp display_script(%__MODULE__{items: [item | stack]}, acc) when is_integer(item) do
+    if item > 0 and item < 0x4C do
+      display_script(%__MODULE__{items: stack}, acc <> " OP_PUSHBYTES_#{item}")
+    else
+      {:ok, op_atom} = get_op_atom(item)
+      upper_op = op_atom |> to_string() |> String.upcase()
+      display_script(%__MODULE__{items: stack}, acc <> " " <> upper_op)
+    end
+  end
+
+  defp display_script(%__MODULE__{items: [item | stack]}, acc) when is_binary(item) do
+    display_script(%__MODULE__{items: stack}, acc <> " " <> Base.encode16(item, case: :lower))
+  end
+
+  # SCRIPT TYPE DETERMINERS
+
+  @doc """
+  	is_p2pk? returns whether a given script is of the p2pk format:
+  	<33-byte or 65-byte pubkey> OP_CHECKSIG
+  """
+  @spec is_p2pk?(t()) :: boolean
+  def is_p2pk?(script) do
+    try do
+      {:ok, len, script} = pop(script)
+      {:ok, pubkey, script} = pop(script)
+      {:ok, 0xAC, script} = pop(script)
+
+      len == byte_size(pubkey) and byte_size(pubkey) in [33, 65] and empty?(script)
+    rescue
+      _ in MatchError -> false
+    end
+  end
+
+  @doc """
+  	is_p2pkh? returns whether a given script is of the p2pkh format:
+  	OP_DUP OP_HASH160 OP_PUSHBYTES_20 <20-byte hash> OP_EQUALVERIFY OP_CHECKSIG
+  """
+  @spec is_p2pkh?(t()) :: boolean
+  def is_p2pkh?(script) do
+    try do
+      {:ok, 0x76, script} = pop(script)
+      {:ok, 0xA9, script} = pop(script)
+      {:ok, @h160_length, script} = pop(script)
+      {:ok, <<_::binary-size(@h160_length)>>, script} = pop(script)
+      {:ok, 0x88, script} = pop(script)
+      {:ok, 0xAC, script} = pop(script)
+      empty?(script)
+    rescue
+      _ in MatchError -> false
+    end
+  end
+
+  @doc """
+  	is_p2sh? returns whether a given script is of the p2sh format:
+  	OP_HASH160 OP_PUSHBYTES_20 <20-byte hash> OP_EQUAL
+  """
+  @spec is_p2sh?(t()) :: boolean
+  def is_p2sh?(script) do
+    try do
+      {:ok, 0xA9, script} = pop(script)
+      {:ok, @h160_length, script} = pop(script)
+      {:ok, <<_::binary-size(@h160_length)>>, script} = pop(script)
+      {:ok, 0x87, script} = pop(script)
+      empty?(script)
+    rescue
+      _ in MatchError -> false
+    end
+  end
+
+  @doc """
+  	is_p2wpkh? returns whether a given script is of the p2wpkh format:
+  	OP_0 OP_PUSHBYTES_20 <20-byte hash>
+  """
+  @spec is_p2wpkh?(t()) :: boolean
+  def is_p2wpkh?(script) do
```

Comment:
https://github.com/SachinMeier/bitcoinex/pull/5

I made a simple for this. Please let me know your thought